synapse/rest/client/v2_alpha/devices.py

```diff
@@ -45,6 +45,52 @@ def on_GET(self, request):
         )
         defer.returnValue((200, {"devices": devices}))
 
+class DeleteDevicesRestServlet(servlet.RestServlet):
+    PATTERNS = client_v2_patterns("/delete_devices", releases=[], v2_alpha=False)
+
+    def __init__(self, hs):
+        """
+        Args:
+            hs (synapse.server.HomeServer): server
+        """
+        super(DeleteDevicesRestServlet, self).__init__()
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.device_handler = hs.get_device_handler()
+        self.auth_handler = hs.get_auth_handler()
+
+    @defer.inlineCallbacks
+    def on_POST(self, request):
+        try:
+            body = servlet.parse_json_object_from_request(request)
+
+        except errors.SynapseError as e:
+            if e.errcode == errors.Codes.NOT_JSON:
+                # deal with older clients which didn't pass a J*DELETESON dict
+                # the same as those that pass an empty dict
+                body = {}
+            else:
+                raise
```

Comment:
Generally its best to do `raise e`, as if you later add a `yield foo` then you will re-raise the wrong error